service/controller/azuremachinepool/handler/nodepool/create_deployment_uninitialized.go

```diff
@@ -94,12 +95,13 @@ func (r *Resource) deploymentUninitializedTransition(ctx context.Context, obj in
 	// Compute desired state for Azure ARM Deployment.
 	desiredDeployment, err := r.getDesiredDeployment(ctx, storageAccountsClient, release, machinePool, &azureMachinePool, azureCluster, vmss)
 	if IsNotFound(err) {
-		r.Logger.Debugf(ctx, "Azure resource not found", "stack", microerror.JSON(err))
+		r.Logger.Debugf(ctx, "Azure resource not found")
 		r.Logger.Debugf(ctx, "canceling resource")
 		return currentState, nil
 	} else if IsSubnetNotReadyError(err) {
-		r.Logger.Debugf(ctx, "subnet is not Ready, it's probably still being created", "stack", microerror.JSON(err))
+		r.Logger.Debugf(ctx, "subnet is not Ready, it's probably still being created")
 		r.Logger.Debugf(ctx, "canceling resource")
+		r.Logger.Debugf(ctx, microerror.JSON(err))
```

Comment:
In this case, it may be useful to see what happened, that's why I tried logging the error anyway. WDYT?

Reply:
Would it make sense to log it before `canceling resource` ?

Reply:
Done